src/Controller/Organization/CommissionableAsset/AssetAddController.php

```diff
@@ -7,24 +7,45 @@
 use App\Entity\CommissionableAsset;
 use App\Entity\Organization;
 use App\Form\Type\CommissionableAssetType;
+use App\Repository\AssetTypeRepository;
 use App\Security\Voter\OrganizationVoter;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\Exception\InvalidParameterException;
 
 /**
  * @Route("/add", name="app_organization_asset_add", methods={"GET", "POST"})
  * @IsGranted(OrganizationVoter::CAN_ADD_ASSET, subject="organization")
  */
 class AssetAddController extends AbstractController
 {
+    private AssetTypeRepository $assetTypeRepository;
+
+    public function __construct(AssetTypeRepository $assetTypeRepository)
+    {
+        $this->assetTypeRepository = $assetTypeRepository;
+    }
+
     public function __invoke(Request $request, Organization $organization): Response
     {
+        /** @var Organization $parentOrganization */
+        $parentOrganization = $organization->isParent() ? $organization : $organization->parent;
+
+        $assetType = null;
+        if ($request->query->has('type')) {
+            $assetType = $this->assetTypeRepository->findByOrganizationAndId($parentOrganization, (int) $request->query->get('type'));
```

Comment:
why not use `$organization->getParent()` in `AssetTypeRepository::findByOrganizationAndId()` ? this way we can get the asset type in a more generic way